src/main/java/com/github/tomakehurst/wiremock/common/xml/XmlNode.java

```diff
@@ -39,14 +39,12 @@
 
 public class XmlNode {
 
-  protected static final InheritableThreadLocal<XPath> XPATH_CACHE =
-      new InheritableThreadLocal<>() {
-        @Override
-        protected XPath initialValue() {
-          final XPathFactory xPathfactory = XPathFactory.newInstance();
-          return xPathfactory.newXPath();
-        }
-      };
+  protected static final ThreadLocal<XPath> XPATH_CACHE =
+      ThreadLocal.withInitial(
+          () -> {
+            final XPathFactory xPathfactory = XPathFactory.newInstance();
+            return xPathfactory.newXPath();
+          });
 
   protected static final InheritableThreadLocal<Transformer> TRANSFORMER_CACHE =
```

Comment:
Can we make the same change here too? Guessing this is also at risk from the same issue.

Reply:
Yes let's do it. Transformer class is not thread-safe also according documentation